services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainMetrics.java

```diff
@@ -162,15 +175,57 @@ public BeaconChainMetrics(
             TekuMetricCategory.BEACON,
             "previous_correct_validators",
             "Number of validators who voted for correct source and target checkpoints in the previous epoch");
+
+    currentEpochParticipationWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "current_epoch_participation_weight",
+            "Total effective balance of all validators who voted for correct source and target checkpoints in the current epoch");
+    previousEpochParticipationWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "previous_epoch_participation_weight",
+            "Total effective balance of all validators who voted for correct source and target checkpoints in the previous epoch");
+
+    currentEpochTotalWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "current_epoch_total_weight",
+            "Total effective balance of all active validators in the current epoch");
+    previousEpochTotalWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "previous_epoch_total_weight",
+            "Total effective balance of all active validators in the previous epoch");
   }
 
   @Override
   public void onSlot(final UInt64 slot) {
-    recentChainData.getChainHead().ifPresent(this::updateMetrics);
+    recentChainData.getChainHead().ifPresent(head -> updateMetrics(slot, head));
   }
 
-  private void updateMetrics(final StateAndBlockSummary head) {
+  private void updateMetrics(final UInt64 slot, final StateAndBlockSummary head) {
     final BeaconState state = head.getState();
```

Comment:
do we need to make sure the StateAndBlockSummary are current epoch?  If we change epoch and there's an empty slot it might not match?

Reply:
Hmm, good question.  It won't have updated if there's an empty slot but it will then update when the next block is received so it's probably not a big deal.  The alternative would be to use `retrieveStateAtSlot` but that risks triggering the processing of a bunch of empty slots and epoch transitions just to update metrics which I don't think we want to do.

So I'd probably leave it as is I think...